### PR TITLE
Add style lint to site

### PR DESCRIPTION
**Pull request by SebiVPS**

### CODE REVIEW

File: site/.stylelintrc.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "extends": "stylelint-config-standard",
+    "customSyntax": "postcss-styled-syntax",
+    "rules": {
+        "no-invalid-double-slash-comments": null,
+        "block-no-empty": true,
+        "declaration-empty-line-before": null,
+        "declaration-block-no-redundant-longhand-properties": null
+    }
```
Comment (johnnyomair) [inline site/.stylelintrc.json:9]: Could you please explain for each rule why you're disabling it?

**Review by johnnyomair — COMMENTED** [review]
I think it's a good idea to add this!

File: site/src/pages/_app.page.tsx [hunk 2]
```diff
@@ -77,7 +78,8 @@ const GlobalStyle = createGlobalStyle`
     }
 
     /* Create a root stacking context: https://www.joshwcomeau.com/css/custom-css-reset/#eight-root-stacking-context-9 */
-    #root,
+    /* stylelint-disable selector-id-pattern */
+    #root,
```
Comment (jamesricky) [inline site/src/pages/_app.page.tsx:82]: I don't think we should prevent ID selectors. 
Especially, if we need this one by default in every project.

Comment (nsams) [reply site/src/pages/_app.page.tsx:82]: It's a good idea to never use id selectors, and it should be linted. Exceptions apply, and for those the rule is disabled.

**Comment by nsams** [comment]
Have you considered a eslint based plugin?

**Comment by SebiVPS** [comment]
> Have you considered a eslint based plugin?

Do you have one in your mind? 
@martinrohrmoser did the research, maybe you considered a eslint based plugin?

**Comment by nsams** [comment]
> Do you have one in your mind?

no, I'm just the one that questions everything :)

**Review by mariokemetinger — APPROVED** [review]

**Review by martinrohrmoser — APPROVED** [review]

**Review by dkarnutsch — APPROVED** [review]

**Comment by johnnyomair** [comment]
> Have you considered a eslint based plugin?

I did some research on how to lint css with eslint, but it seems that this is only possible for the most basic things (e.g. formatting). I also haven't found a plugin that incorporates stylelint into eslint. It seems to me that using stylelint standalone is the wat to go.

**Review by johnnyomair — APPROVED** [review]

**Comment by johnnyomair** [comment]
I'd like to wait on @jamesricky's approval here

**Review by thomasdax98 — APPROVED** [review]

**Review by jamesricky — APPROVED** [review]
I think we should add this to admin as well for consistency

**Comment by johnnyomair** [comment]
> I think we should add this to admin as well for consistency. @johnnyomair @thomasdax98 what do you think?

Do we have that much custom CSS in the admin? But yes, we could do that in a follow-up PR.

**Pull request merged**